gtest_parallel.py

```diff
@@ -436,6 +436,57 @@ def add_stats(stats, task, idx):
   def flush(self):
     self.out.flush_transient_output()
 
+def join_gtest_output_files(files, output_filename):
+  '''join given files into a single one'''
+  from xml.dom import minidom
+
+  files = [f for f in files if os.path.exists(f) and os.path.getsize(f) > 0]
```

Comment:
Can you insert a TODO to revisit how empty files / crashes are handled and highlight that we skip that for now?

Also could we make sure to print a summary of how many tests failed to generate output? I think this may go through list of files before stripping out the !exists/empty ones. It would be OK to print every single failure here I think and not necessarily print x/y tests aborted. Can we get the test name corresponding to the failure as well?